src/map/entities/mobentity.cpp

```diff
@@ -669,6 +669,17 @@ void CMobEntity::OnMobSkillFinished(CMobSkillState& state, action_t& action)
             PTarget->PAI->EventHandler.triggerListener("WEAPONSKILL_TAKE", PTarget, this, PSkill->getID(), state.GetSpentTP(), &action);
         }
 
+        if (objtype == TYPE_PET && PMaster && PMaster->objtype == TYPE_PC )
+        {
+            auto mob = dynamic_cast<CMobEntity *>(PTarget);
+            if (mob && mob->objtype == TYPE_MOB && !mob->CalledForHelp())
```

Comment:
If it casts to a CMobEntity, we already know it's a TYPE_MOB. Don't need to do both checks.

Reply:
ha, totally fair, if its not a mob dynamic cast would return nullptr, right?

Reply:
Correct!

Reply:
I'm also not sure if this should even check for TYPE_PET, just if it has a PC master, in case of a charmed mob killing an unclaimed link with an aoe mobskill.